java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -305,14 +309,20 @@ int maxSteps() {
   }
 
   private void execute(String signature, SquidClassLoader classLoader) {
-    programState = ProgramState.EMPTY_STATE;
-    steps = 0;
-    BytecodeCFGBuilder.BytecodeCFG bytecodeCFG = BytecodeCFGBuilder.buildCFG(signature, classLoader);
-    if (bytecodeCFG == null) {
+    BytecodeCFGMethodVisitor cfgVisitor = new BytecodeCFGMethodVisitor();
+    MethodLookup lookup = MethodLookup.lookup(signature, classLoader, cfgVisitor);
+    if (lookup == null) {
+      LOG.debug("Method lookup failed for %s", signature);
+      return;
+    }
+    methodBehavior.setDeclaredExceptions(lookup.declaredExceptions);
+    methodBehavior.setVarArgs(lookup.isVarArgs);
+    BytecodeCFG bytecodeCFG = cfgVisitor.getCfg();
+    if (bytecodeCFG == null || methodIsBlacklisted(signature)) {
```

Comment:
shouldn't the black listing be checked _before_ going for the lookup ? This is basically the first thing this method should do (it is validating the arguments).

Reply:
the idea is that even if method is blacklisted we want to use the thrown exceptions. This is especially important for methods from `java.lang.Class` , because it's quite common to catch `NoSuchMethodException` and similar.

Reply:
So maybe we should anyway avoid computing the CFG in that case ?

Reply:
Indeed, see next commit